gradio/upload.py

```diff
@@ -0,0 +1,173 @@
+import os
+import random
+import huggingface_hub
+import json
+
+
+def get_existing_configuration(readme_file):
```

Comment:
Can be replaced to use `huggingface_hub`'s parsing method `metadata_load`:

```py
>>> from huggingface_hub import metadata_load
>>> metadata_load("/home/wauplin/projects/git-vs-http-benchmark/README.md")
{'title': 'Git Vs Http Benchmark', 'emoji': '🐢', 'colorFrom': 'indigo', 'colorTo': 'blue', 'sdk': 'gradio', 'sdk_version': '3.17.0', 'app_file': 'app.py', 'pinned': False}
```

You also have `metadata_save` to save metadata back to a README file while keeping the existing "raw" content. Both methods should be robust to special cases (like \r\n line break on Windows).